src/shared/pcre2-util.c

```diff
@@ -33,12 +35,124 @@ int dlopen_pcre2(void) {
                         DLSYM_ARG(pcre2_get_error_message),
                         DLSYM_ARG(pcre2_match),
                         DLSYM_ARG(pcre2_get_ovector_pointer));
+#else
+        return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "PCRE2 support is not compiled in.");
```

Comment:
the dlopen_many_sym_or_warn() call will log at LOG_ERR level, but here you only log at debug level. The function should log on the same level in either case.

Reply:
It logs at LOG_ERR if a symbol cannot be found, but it only logs at LOG_DEBUG if the library is not installed, so the case is not as clear cut. Similarly, dlopen_bpf() also only logs at debug level if BPF support was not compiled in.

I think we shouldn't log at higher level than DEBUG when checking if a library is available or compiled in, especially since we might call a function like this to figure out if a particular library is available or not. That would lead to a lot of log spam overall. That's why here I think we should deviate from either always logging or never logging, since the logging when as symbol is missing in dlopen_many_sym_or_warn() is very useful and we should keep that.